### PR TITLE
Disable building tests by default

### PR DESCRIPTION
**Pull request by riordant**
Currently tests are built by default where the host is Linux-based. This PR disables building tests by default. 
We also have a configure flag to disable tests, `--disable-tests`. Given the new functionality, I propose we change this to `--enable-tests`. This is also updated here.
Fixes https://github.com/zcoinofficial/zcoin/issues/261

### CODE REVIEW

File: configure.ac [hunk 1]
```diff
@@ -106,17 +106,10 @@ AC_ARG_ENABLE([upnp-default],
...
-     zusetests=yes
-  ;;
-esac
-
 AC_ARG_ENABLE(tests,
-    AS_HELP_STRING([--disable-tests],[do not compile tests (default is to compile)]),
-    [use_tests=$enableval],
-    [use_tests=$zusetests])
+    AS_HELP_STRING([--enable-tests],[compile tests (default is not to compile)]),
+    [use_tests=$withval],
```
Comment (ultimaweapon) [inline configure.ac]: Seem like `$withval` should be using with `--with-xxx` instead. For `--enable-xxx` it look like we need to use `$enableval` instead.

**Comment by a-bezrukov** [comment]
Disabled the tests, made `make check` error when tests were disabled and did a minor refactoring.

File: src/Makefile.am [hunk 2]
```diff
@@ -613,6 +613,9 @@ endif
 
 if ENABLE_TESTS
 include Makefile.test.include
+else
+check-local:
```
Comment (ultimaweapon) [inline src/Makefile.am]: Not sure if this going to run first when we run `make check`. I take a look at `check-local` in `src/Makefile.test.include` and compare with test output found out it run after all unit tests have been ran.

Comment (a-bezrukov) [reply src/Makefile.am]: Makefile.test.include is not included when we configured without tests. This check-local purpose is to fail the tests when we configured without tests.

Comment (riordant) [reply src/Makefile.am]: @ultimaweapon Are you ok with Andrey's reply?

**Pull request merged**